### DISCUSSION TITLE
Make fish a runtime, only include core features for scripting?

### DISCUSSION THREAD

**Discussion by buddhi-deep** [fish-shell/fish-shell#9465] [Ideas]
I'm wondering if this could make fish less dependent

**Comment by faho** [comment]
You basically asked this before in #8867, and [the answer](https://github.com/fish-shell/fish-shell/discussions/8867#discussioncomment-2544005) remains the same:

No, it's not possible, you *can't* just remove features - e.g. `read --shell` includes the entire reader functionality. Even if you were to remove the entire reader (say you make a "script fish" that just can't `read --shell`), that wouldn't really *help* much. It would make fish very slightly smaller (but a full stripped fish binary is ~2M, so this might be ~1.7M), but it wouldn't make it "less dependent" - you would *still* need to link to ncurses for other things like `set_color` and output logic for a bunch of things (like `string pad` or `string shorten`).

Fish uses three libraries: pcre2 (mostly for `string`'s regex support), ncurses (for anything to do with input/output and other terminal specifics) and the C/C++ standard library. You won't be able to remove any of the three without creating something that isn't woefully incompatible to actual fish.

The best we can do is an appimage (#6475). I built one: https://github.com/faho/fish-shell/releases/tag/fish-3.6.0-x86_64, you can try it out, it should work on linux with glibc >= CentOS 7.

**Reply by buddhi-deep** [reply]
Thanks for replying me this question Carefully even if I have asked a similar question (I didn't notice that)
Sorry for waste your time
👍

**Comment by buddhi-deep** [comment]
Ok so i've just made a auto builder for static ver
So that system with older fish doesn't need to install a newer version to run new scripts
